guardian-ui/src/GuardianContext.tsx

```diff
@@ -0,0 +1,346 @@
+import React, {
+  createContext,
+  Dispatch,
+  ReactNode,
+  useCallback,
+  useEffect,
+  useReducer,
+  useState,
+} from 'react';
+import {
+  SetupState,
+  SetupAction,
+  SETUP_ACTION_TYPE,
+  SetupProgress,
+  ConfigGenParams,
+  ConsensusState,
+  ServerStatus,
+} from './types';
+import { ApiInterface, GuardianApi } from './GuardianApi';
+import { JsonRpcError } from 'jsonrpc-client-websocket';
+
+const LOCAL_STORAGE_KEY = 'guardian-ui-state';
```

Comment:
TBH I would like to get the password storage out of the API file altogether and move it in here. I'd much prefer that to just be a dumb class that takes all its inputs explicitly, rather than having some magic internal stuff going on.